packages/http/src/types.ts

```diff
@@ -35,7 +35,20 @@ export type HttpAuth =
   | BearerAuth
   | ApiKeyAuth<ApiKeyLocation, string>
   | Oauth2Auth<OAuth2Flow[]>
-  | OpenIDConnectAuth;
+  | OpenIDConnectAuth
+  | NoAuth;
+
+export interface HttpAuthReference {
+  /*
+   * Unique id of the authentication scheme among all schemes.
+   * `undefined` means lack of auth
+   */
+  id?: string;
```

Comment:
Would it make sense to maybe just connect to the actual `HttpAuth` object. I am not sure we would really care if this data structure is serializable as json.

Reply:
Yep, done